azure-kusto-ingest/azure/kusto/ingest/ingest_client.py

```diff
@@ -108,3 +125,32 @@ def ingest_from_blob(self, blob_descriptor: BlobDescriptor, ingestion_properties
         content = ingestion_blob_info_json
         queue_client = queue_service.get_queue_client(queue=random_queue.object_name, message_encode_policy=TextBase64EncodePolicy())
         queue_client.send_message(content=content)
+
+    def validate_endpoint_service_type(self):
+        if not hasattr(self, "_endpoint_service_type") or len(self._endpoint_service_type) == 0 or self._endpoint_service_type.isspace():
```

Comment:
not self._endpoint_service_type.strip()

Reply:
@yihezkel , you should probably set this value in __init__ to a default (e.g. "" or None) to simplify this line.

Reply:
Are you sure? This is my first Python change, so I looked this up. The guidance was that if the variable might not exist (which is the case here), to use hasattr().
And if I use strip and the variable doesn't exist, won't it throw an exception?

Reply:
I agree its better to init the parameters before and avoid those checks.
Is it even reasonable to get a whitespace here? If not you can remove the split and check only empty string, but you use "or" so you shouldn't run into exception here because the first condition will fail the check.

Reply:
@ArielYehezkely 
Yes the way I wrote it, it wouldn't fail because of the "hasattr".
I was responding to your suggestion to replace that with "not self._endpoint_service_type.strip()", which would fail if it wasn't defined.

"Is it even reasonable to get a whitespace here?"
My coding style places an emphasis on resilience. Right now, it can't have whitespace, but perhaps it could in the future. This method is a black box that should produce the correct output given any reasonable input. It's similar to (I presume) why in our C# code base we prefer to use "string.IsNullOrWhiteSpace()" wherever possible.

Reply:
@ArielYehezkely 
Sorry, I misunderstood - you were only suggesting replacing "len(self._endpoint_service_type) == 0 or self._endpoint_service_type.isspace()" with "not self._endpoint_service_type.strip()". I'll do that.